code/network/multimsgs.cpp

```diff
@@ -7461,35 +7531,64 @@ void send_homing_weapon_info( int weapon_num )
 		}
 	}
 
+	if (!wp->weapon_flags[Weapon::Weapon_Flags::Multi_Update_Sent] || IS_MAT_NULL(&Objects[wp->objnum].phys_info.ai_desired_orient)) {
+		flags |= HWIF_BIG_UPDATE;
+		wp->weapon_flags.set(Weapon::Weapon_Flags::Multi_Update_Sent);
+	}
+
 	BUILD_HEADER(HOMING_WEAPON_UPDATE);
+	ADD_DATA(flags);
 	ADD_USHORT( Objects[wp->objnum].net_signature );
 	ADD_USHORT( homing_signature );
 	ADD_SHORT( static_cast<short>(s_index) );
-	
+	ADD_VECTOR( wp->homing_pos);
+
+	if (flags & HWIF_BIG_UPDATE) {
+		fix current_lifetime = Missiontime - wp->creation_time;
+		ADD_DATA(current_lifetime);
```

Comment:
You'll have to use `ADD_INT()` instead here to match the data size of `fix`. Unless if will always fit in a smaller byte size in which case you can use that type instead of `fix`